src/Chapter2.hs

```diff
@@ -620,8 +632,13 @@ Write a function that takes elements of a list only on even positions.
 >>> takeEven [2, 1, 3, 5, 4]
 [2,3,4]
 -}
-takeEven = error "takeEven: Not implemented!"
-
+takeEven :: [a] -> [a]
+takeEven xs = takeEven' 0 xs where
+   takeEven' _ [] = []
+   takeEven' i (x:xs) = if even i
+       then x:takeEven' (i+1) xs
+       else takeEven' (i+1) xs
```

Comment:
Nice idea on the helper functions :+1:

A few additional thoughts on your idea: as you have only two states of the operation (odd or even), instead of accumulators of the current position number, you can just use `Bool`. And on each next recursion call, you just give it `not acc`. This will remove the necessity to check `even` each time, which would save a couple of functions call on each interaction.

Alternatively, you could use pattern matching in `takeEven` instead. As you need to take every second element, it is possible to represent with a few pattern matching clauses 🙂